### ISSUE TITLE
[Bug]: Parser not detecting functions that exist in the same file.

### ISSUE THREAD

**Issue by Tedster59** [EliotVU/UnrealScript-Language-Service#191]
### Describe the bug

After sticking with version 0.6.3 or so for a while, I was trying 7.1 and now 8.1, and I'm seeing that the parser is failing to find function references that exist in the same file.  I'm not sure if there's an issue with the project I'm working on being too large or if there is some other issue, such as too many custom macros, or the fact XCOM 2 uses a custom version of UE3 that is more like a UE3.5

Using VSCode 1.91.1 and extension version 0.8.1 right now.

### Screenshots

For example, it's not detecting this function, as well as several that are called as part of this event.
![image](https://github.com/user-attachments/assets/fa97bad9-f6dc-4dde-92bf-f1f710dfc8f5)

Ctrl F for it, the function is definitely here.

![image](https://github.com/user-attachments/assets/610fb887-d530-4c21-b1aa-3999d3ab0c0a)

**Comment by EliotVU** [comment]
Yes, usually caused by in-lined macros that the extension is not aware of.

**Comment by Tedster59** [comment]
Hmm, what's the best way to make it aware of the custom macros? They are defined in an extra_globals.uci file that is in a folder that is part of the vscode workspace.  I've also noticed it sometimes failing to find class definitions that it would before.

**Comment by Tedster59** [comment]
Also potentially related, in the same file, it will find the class definition, but then fail to find it farther down the file.
found, go to definition works, etc.

This class is mentioned 57 times in the file per ctrl-f search, and the first five are correctly identified.
![image](https://github.com/user-attachments/assets/0038387d-ecf4-410c-833c-a3186132da66)

#6 and afterwards are not located and go to definition does nothing.
![image](https://github.com/user-attachments/assets/4d20169e-a297-4503-98bb-3503771df4ab)

Some functions are not underlined red, but Go To Definition doesn't work on them.
![image](https://github.com/user-attachments/assets/cebf73c9-59b7-476c-a1b7-d441149d7608)

**Comment by EliotVU** [comment]
Closing in favor of #10
